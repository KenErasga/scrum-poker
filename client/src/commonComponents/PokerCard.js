import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Collapse from '@material-ui/core/Collapse';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles((theme) => ({
    root: {
        padding: theme.spacing(2),
        textAlign: 'center',
        color: theme.palette.text.secondary,
    },
    expand: {
        transform: 'rotate(0deg)',
        marginLeft: 'auto',
        transition: theme.transitions.create('transform', {
            duration: theme.transitions.duration.shortest,
        }),
    },
    expandOpen: {
        transform: 'rotate(180deg)',
    },
}));

const PokerCard = ({ name, estimate, isExpanded }) => {
    const classes = useStyles();
    return (
        <Card className={classes.root}>
            <CardContent>
                <Typography variant="h4" component="h4">
                    {name}
                </Typography>
            </CardContent>
            <Collapse in={isExpanded} timeout="auto" unmountOnExit>
                <CardContent>
                    <Typography style={{
                        display: "flex",
                        justifyContent: "center"
                    }} variant="h1" component="h1">{estimate}</Typography>
                </CardContent>
            </Collapse>
        </Card>
    );
};

export default PokerCard;
