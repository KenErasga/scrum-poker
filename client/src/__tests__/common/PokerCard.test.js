import React from 'react';
import renderer from 'react-test-renderer';
import PokerCard from '../../common/PokerCard';

describe('PokerCard', () => {

    it('PokerCard renders expanded correctly', () => {
    const tree = renderer
        .create(<PokerCard name='test-name' estimate='5' isExpanded={true}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });

    it('PokerCard renders not expanded correctly', () => {
        const tree = renderer
            .create(<PokerCard name='test-name' estimate='5' isExpanded={false}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});
