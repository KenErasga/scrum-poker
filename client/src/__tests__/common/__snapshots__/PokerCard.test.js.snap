// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokerCard PokerCard renders expanded correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
    >
      test-name
    </h4>
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-entered"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          className="MuiCardContent-root"
        >
          <h1
            className="MuiTypography-root MuiTypography-h1"
            style={
              Object {
                "display": "flex",
                "justifyContent": "center",
              }
            }
          >
            5
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PokerCard PokerCard renders not expanded correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
    >
      test-name
    </h4>
  </div>
</div>
`;
